geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/AccumuloFeatureStore.scala

```diff
@@ -50,6 +52,8 @@ class AccumuloFeatureStore(val dataStore: AccumuloDataStore, val featureName: Na
         val feature = iter.next()
         val newFeature = fw.next()
 
+        minMaxVisitorO.foreach ( _.visit(feature) )
```

Comment:
Rather than calling this option which may be ```None``` for each iteration of the loop, I'd suggest traversing the collection a second time...

```scala
val iter = collection.features()
while(iter.hasNext) {
   val feature = iter.next()
   val newFeature = iter.next()
   ...
   // write feature
}

minMaxVisitorO.foreach { visitor = > collection.features().foreach { f => visitor.visit(f) } }
```